src/js/components/OptionalPortsAndServiceDiscoveryComponent.jsx

```diff
@@ -17,36 +17,35 @@ import Util from "../helpers/Util";
 
 const fieldsetId = "portDefinitions";
 
-function determinePortDefinitionsType(fields) {
+function isBridgeNetworkMode(fields) {
   if (fields.dockerNetwork === ContainerConstants.NETWORK.BRIDGE) {
-    return ContainerConstants.NETWORK.BRIDGE;
-  }
-
-  if ((fields.dockerImage != null && fields.dockerImage !== "") ||
-      fields.dockerNetwork === ContainerConstants.NETWORK.HOST) {
-    return ContainerConstants.NETWORK.HOST;
+    return true;
   }
+  return false;
+}
 
-  return ContainerConstants.TYPE.MESOS;
+function hasDockerImage(fields) {
+  return fields.dockerImage != null && fields.dockerImage !== "";
 }
 
 function isTooComplexStructure(fields, hasVIP = false) {
-  return fields.portDefinitions != null &&
-      fields.portDefinitions.some(portDefinition => {
-        if (!hasVIP && portDefinition.vip != null) {
-          return true;
-        }
+  return isBridgeNetworkMode(fields) ||
```

Comment:
This condition is extremely hard to read (already before your changes), how about to swap the test function out?

```JavaScript
function isTooComplexStructure(fields, hasVIP = false) {
  var areTooComplex = portDefinition => {
    if (!hasVIP && portDefinition.vip != null) {
      return true;
    }

    if (portDefinition.labels != null) {
      let labels = Object.assign({}, portDefinition.labels);
      if (hasVIP) {
        delete labels["VIP_0"];
      }
      return !!Object.keys(labels).length;
    }
    return false;
  };

  return isBridgeNetworkMode(fields) ||
    fields.portDefinitions != null &&
    fields.portDefinitions.some(areTooComplex);
}
```